src/components/SearchBar.tsx

```diff
@@ -55,17 +56,29 @@ const SearchBar: React.FC<SearchBarProps> = (props) => {
   const history = useHistory();
 
   const onChangeHandler = (label: string | null) => {
-    const item = _.find<SearchItem>(data, { label: label ?? "" });
+    const item = find<SearchItem>(data, { label: label ?? "" });
     if (!item) {
       history.push(`/search?label=${label}`);
     } else if (item.isWord) {
       history.push(`/disambiguation?label=${label}`);
     } else {
       const prop = item.items[0];
-      history.push("/term", prop);
+      history.push(generateTermRoute(prop));
     }
   };
 
+  const handleChange = (event: ChangeEvent<{}>, newInputValue: string) => {
+    setInputValue(newInputValue);
+  };
+
+  const debouncedChangeHandler = useMemo(() => debounce(handleChange, 300), []);
```

Comment:
Keep in mind that if you define the debounced handler like this, then it will forever contain only the first instance of `handleChange` that was created during the first render of the component. Every subsequent render creates new `handleChange` variable which is essentially useless. It would be cleaner to move the code inside `useMemo` as well, but it is no big deal.

Reply:
One more thing - you should include `setInputValue` as a dependency of the `useMemo` hook.